optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/api/IntervalBreak.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.common.experimental.api;
+
+/**
+ * An IntervalBreak is a gap between two consecutive interval clusters. For instance,
+ * the list [(1,3),(2,4),(3,5),(7,8)] has a break of length 2 between 5 and 7.
+ *
+ * @param <IntervalType_> The type of value in the sequence
+ * @param <DifferenceType_> The type of difference between values in the sequence
+ */
+public interface IntervalBreak<IntervalType_, PointType_ extends Comparable<PointType_>, DifferenceType_ extends Comparable<DifferenceType_>> {
```

Comment:
Dtto. Maybe we have the same break interface for both the sequence and the cluster?

Reply:
My main issue with this proposal is the naming of the `nextSequence`/`previousSequence` and `nextCluster/previousCluster`. Strictly speaking, Sequences are a specialized version of Clusters where each interval end point is a constant different away from the start point, but they aren't "viewed" as intervals by the users, but rather points. If they both share the same interface, they need to have the same name.

Reply:
That is a good point - but wouldn't that be visible from the generic type? In one instance, `getNext()` would return `Sequence`, in another it would return `Cluster`.

That said, I don't think this is an absolute necessity.

Reply:
`getNext()` for me, would mean "get next break" not "get the thing after the break"